### ISSUE TITLE
Fix the problem with grouping users where consider_alcohol is true

### ISSUE THREAD

**Issue by TitusKirch** [fsr5-fhaachen/portals#326]
Last year the groups were unequal in size because there was a logic problem with the division of consider_alcohol true users.

**Comment by BenediktHaas96** [comment]
Tl;Dr: assignLeftover only targeted groups with non-drinkers for leftover assignments, fixed by small logic change

Expected Behavior:
If the division detects that it needs to consider drinking habits, assignLeftover should assign leftover non-drinkers only to groups with non-drinkers. Drinkers should be assigned to any group in such a way, that groups are on average equal in size.

Actual Behavior:
assignLeftover did not detect if there were any non-drinkers left in the leftover registrations and assigned ALL leftovers to non-drinker groups only, as long as the division needed to consider drinking habits.

Fix:
1) Disable assigning only to non-drinker groups as soon as leftovers contain no more non-drinkers
2) If currently handled leftover registration is a drinker, assign them to any open group with least members
